static/js/user_groups_settings_ui.js

```diff
@@ -95,6 +103,36 @@ function show_right_section() {
     $(".user-groups-header").addClass("slide-left");
 }
 
+export function add_group_to_table(group) {
+    if (is_group_already_present(group)) {
+        // If a group is already listed/added in groups modal,
+        // then we simply return.
+        // This can happen in some corner cases (which might
+        // be backend bugs) where a realm administrator may
+        // get two user_group-add events.
+        return;
+    }
+
+    const settings_html = render_user_group_settings({
+        group,
+        can_edit: user_group_edit.can_edit(group.id),
+    });
+
+    group_list_widget.replace_list_data(user_groups.get_realm_user_groups());
+    ui.get_content_element($("#manage_groups_container .settings")).append($(settings_html));
+
+    // TODO: Address issue for visibility of newely created group.
+    if (user_group_create.get_name() === group.name) {
+        // This `user_group_create.get_name()` check tells us whether the
+        // group was just created in this browser window; it's a hack
+        // to work around the server_events code flow not having a
+        // good way to associate with this request because the group
+        // ID isn't known yet.
```

Comment:
We can possibly borrow the `queue_id+local_id` API approach we use for sending messages -- basically the client sends a token in the request, and the server arranges to send the `local_id` value back to the client with that `queue_id`.

I don't remember how complicated that was to plumb.

Reply:
Grep for `sender_queue_id` to see how that code path works.